tools/dependencies/README.md

```diff
@@ -40,3 +40,349 @@ This issue appeared in the OSX build with XCode version 8.0 above (reproduced on
+#
+# Do you want to install a symbolic link at /usr/local/cuda?
+# (y)es/(n)o/(q)uit: y
+#
+# Install the CUDA 10.0 Samples?
+# (y)es/(n)o/(q)uit: n
+
+# Set LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=/usr/local/cuda/lib64:${LD_LIBRARY_PATH}
+
+# Check installation
+nvidia-smi
+
+# cuDNN Setup 
+# Take cuDNN 7.5.0 with CUDA 10 for example
+# https://developer.nvidia.com/rdp/cudnn-download
+# Register with NVIDIA and download cudnn-10.0-linux-x64-v7.5.0.56.tgz
+# scp it to your instance
+# https://docs.nvidia.com/deeplearning/sdk/cudnn-install/index.html
+tar -xvzf cudnn-10.0-linux-x64-v7.5.0.56.tgz
+sudo cp cuda/include/cudnn.h /usr/local/cuda/include
+sudo cp cuda/lib64/libcudnn* /usr/local/cuda/lib64
+sudo chmod a+r /usr/local/cuda/include/cudnn.h /usr/local/cuda/lib64/libcudnn*
+# Check cuDNN version
+cat /usr/local/cuda/include/cudnn.h | grep CUDNN_MAJOR -A 2 
+# #define CUDNN_MAJOR 7
+# #define CUDNN_MINOR 5
+# #define CUDNN_PATCHLEVEL 0
+# --
+# #define CUDNN_VERSION (CUDNN_MAJOR * 1000 + CUDNN_MINOR * 100 + CUDNN_PATCHLEVEL)
+#
+# #include "driver_types.h"
+
+# install NCCL
+# take NCCL 2.4.2 for example
+# https://developer.nvidia.com/nccl/nccl2-download-survey
+# Register with NVIDIA and download nccl-repo-ubuntu1604-2.4.2-ga-cuda10.0_1-1_amd64.deb
+sudo dpkg -i nccl-repo-ubuntu1604-2.4.2-ga-cuda10.0_1-1_amd64.deb
+sudo apt-key add /var/nccl-repo-2.4.2-ga-cuda10.0/7fa2af80.pub
+sudo apt update
+sudo apt install libnccl2 libnccl-dev
+# we will check the NCCL version later
+```
+#### 2. Build 
+```
+# Clone MXNet repo
+git clone --recursive https://github.com/apache/incubator-mxnet.git
+cd incubator-mxnet
+# Make sure you pin to specific commit for all the performance sanity check to make fair comparison
+# Make corresponding change on tools/setup_gpu_build_tools.sh
+# to upgrade CUDA version, please refer to PR #14887.
+# Make sure you add new makefile and right debs CUDA uses on the website
+# http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/
+
+# Build PyPi package
+tools/staticbuild/build.sh cu100mkl pip
+
+# Wait for 10 - 30 mins, you will find libmxnet.so under the incubator-mxnet/lib
+
+# Install python frontend
+cd python
+pip install -e . --pre
+# Test MXNet
+>>> import mxnet as mx
+>>> mx.nd.ones((2, 5) ctx=mx.gpu(0))
+>>> exit()
+
+# Test NCCL version
+export NCCL_DEBUG=VERSION
+vim tests/python/gpu/test_nccl.py
+# Remove @unittest.skip("Test requires NCCL library installed and enabled during build") then run
+nosetests --verbose tests/python/gpu/test_nccl.py
+# test_nccl.test_nccl_pushpull ... NCCL version 2.4.2+cuda10.0
+# ok
+# ----------------------------------------------------------------------
+# Ran 1 test in 67.666s
+
+OK
+```
+#### 3. Performance Sanity Check
+We will test against 3 basic models
+###### ResNet50 with ImageNet
+```
+# Download the ImageNet on http://image-net.org/download and make record file
+# Install prerequisite package
+pip2 install psutil --user
```

Comment:
Why pip2?

Reply:
our benchmark script is written in python2